### PR TITLE
Add generic support for mpd manifests(dash formats)

### PR DESCRIPTION
**Pull request by remitamine**
the implementation is based on the information from the ISO/IEC 23009-1:2014.
i added support for vevo.
it can also the segments from kaltura and bbc and concatenate the audio segments.
but there is a problem with video concatenation:
for bbc ffmpeg complain about an unsupported codec.
and for kaltura it said: `could not find corresponding track id 1`

### CODE REVIEW

File: youtube_dl/extractor/common.py [hunk 1]
```diff
@@ -1330,81 +1332,145 @@ def _parse_xspf(self, playlist, playlist_id):
...
-        return self._parse_dash_manifest(
-            dash_doc, namespace=namespace, formats_dict=formats_dict)
+    def _parse_mpd(self, mpd_doc, mpd_id=None, mpd_base_url='', formats_dict={}):
+        if mpd_doc.get('type') == 'dynamic':
+            return []
 
-    def _parse_dash_manifest(self, dash_doc, namespace=None, formats_dict={}):
-        def _add_ns(path):
-            return self._xpath_ns(path, namespace)
+        def extract_multisegment_info(element, ms_parent_info):
```
Comment (yan12125) [inline youtube_dl/extractor/common.py]: I guess this function is too long to be a nested function.

File: youtube_dl/extractor/common.py [hunk 2]
```diff
@@ -1330,81 +1332,145 @@ def _parse_xspf(self, playlist, playlist_id):
...
+            ms_info = ms_parent_info.copy()
+            segment_list = element.find(self._xpath_ns('SegmentList', namespace))
```
Comment (yan12125) [inline youtube_dl/extractor/common.py]: Any reason for not using helper functions like ```_add_ns```? Codes look lengthy without that.

**Comment by yan12125** [comment]
Thanks for this work. My previous work on Facebook is not done yet. At least two problems should be fixed:

1. Default ```format``` for FacebookIE should be ```bestvideo+bestaudio/best```
2. DASH formats should have lower priorities

I'll fix them after this PR merged.

File: youtube_dl/extractor/common.py (hunk 1 above)
Comment (remitamine) [inline youtube_dl/extractor/common.py]: but it is used only in _parse_mpd and it too specific(extract multisegment info)

File: youtube_dl/extractor/common.py [hunk 3]
```diff
@@ -1330,81 +1332,145 @@ def _parse_xspf(self, playlist, playlist_id):
...
+                            'format_id': mpd_id or representation_id,
+                            'url': base_url,
+                            'width': int_or_none(representation_attrib.get('width')),
+                            'height': int_or_none(representation_attrib.get('height')),
+                            'tbr': int_or_none(representation_attrib.get('bandwidth'), 1000),
+                            'asr': int_or_none(representation_attrib.get('audioSamplingRate')),
+                            'fps': int_or_none(representation_attrib.get('frameRate')),
+                            'vcodec': 'none' if content_type == 'audio' else representation_attrib.get('codecs'),
+                            'acodec': 'none' if content_type == 'video' else representation_attrib.get('codecs'),
+                            'language': representation_attrib.get('lang'),
```
Comment (yan12125) [inline youtube_dl/extractor/common.py]: test:facebook_2 now returns:
```
$ youtube-dl -vF test:facebook_2                                              
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-vF', 'test:facebook_2']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.02.01
[debug] Git HEAD: 1df4141
[debug] Python version 3.5.1 - Linux-4.4.0-4-ARCH-x86_64-with-arch-Arch-Linux
[debug] exe versions: avconv v12_dev0-2305-g77a44f5, avprobe v12_dev0-2305-g77a44f5, ffmpeg 2.8.5, ffprobe 2.8.5, rtmpdump 2.4
[debug] Proxy map: {}
[TestURL] Test URL: https://www.facebook.com/video.php?v=957955867617029
[facebook] 957955867617029: Downloading webpage
[info] Available formats for 957955867617029:
format code                      extension  resolution note
progressive_sd_src               mp4        unknown    
progressive_sd_src_no_ratelimit  mp4        unknown    
485721114944025ad                mp4        audio only [und]DASH audio   49k , mp4a.40.29 (44100Hz)
584741191674072vd                mp4        240x426    [und]DASH video  283k , avc1.4d401e, video only
474346612748657v                 mp4        360x640    [und]DASH video  331k , avc1.64001f, video only (best)
```
Maybe not include ```language``` if it's ```und```?

File: youtube_dl/extractor/vevo.py [hunk 4]
```diff
@@ -197,8 +197,14 @@ def _real_extract(self, url):
...
 
-                if '.mpd' in version_url or '.ism' in version_url:
+                if '.ism' in version_url:
                     continue
+                elif '.mpd' in version_url:
+                    formats.extend(self._extract_mpd_formats(
+                        version_url, video_id, mpd_id='dash-%s' % version,
+                        note='Downloading %s MPD information' % version,
+                        errnote='Failed to download %s MPD information' % version,
+                        fatal=False))
```
Comment (yan12125) [inline youtube_dl/extractor/vevo.py]: It would be better to have a test case.

File: youtube_dl/extractor/common.py (hunk 2 above)
Comment (remitamine) [inline youtube_dl/extractor/common.py]: it's my fault when i merged the code i replaced it directly instead of looking at it first.
i will fix it now.

**Comment by yan12125** [comment]
If you don't mind, could you commit/push changes to ```bbc.py``` and ```kaltura.py``` and give some example webpage URLs?

**Comment by remitamine** [comment]
i also think that the dash download should be changed to extend FragmentFD.

File: youtube_dl/extractor/common.py (hunk 3 above)
Comment (remitamine) [inline youtube_dl/extractor/common.py]: i don't know if it is general or specific to facebook.

File: youtube_dl/extractor/vevo.py (hunk 4 above)
Comment (remitamine) [inline youtube_dl/extractor/vevo.py]: the last video in test cases has dash formats.

**Comment by yan12125** [comment]
> i also think that the dash download should be changed to extend FragmentFD.

It's great but please leave it in another PR. This PR already contains lots of changes.

**Comment by remitamine** [comment]
> If you don't mind, could you commit/push changes to bbc.py and kaltura.py and give some example webpage URLs?

i didn't add support for kaltura and bbc because the download only work for audio.
for videos it can download the segements but the result file has the problems that i already point at.
i can put the code to enable the dash formats here.
kaltura:
```diff
diff --git a/youtube_dl/extractor/kaltura.py b/youtube_dl/extractor/kaltura.py
index ccbc39c..0720fb4 100644
--- a/youtube_dl/extractor/kaltura.py
+++ b/youtube_dl/extractor/kaltura.py
@@ -165,8 +165,13 @@ class KalturaIE(InfoExtractor):
             m3u8_url += '?referrer=%s' % referrer
         formats.extend(self._extract_m3u8_formats(
             m3u8_url, entry_id, 'mp4', 'm3u8_native', m3u8_id='hls', fatal=False))
+        mpd_url = info['dataUrl'].replace('format/url', 'format/mpegdash')
+        if referrer:
+            mpd_url += '?referrer=%s' % referrer
+        formats.extend(self._extract_mpd_formats(
+            mpd_url, entry_id, fatal=False))
 
-        self._check_formats(formats, entry_id)
+        #self._check_formats(formats, entry_id)
         self._sort_formats(formats)
 
         return {
```
bbc:
```diff
diff --git a/youtube_dl/extractor/bbc.py b/youtube_dl/extractor/bbc.py
index 6ddee68..83773e1 100644
--- a/youtube_dl/extractor/bbc.py
+++ b/youtube_dl/extractor/bbc.py
@@ -246,7 +246,8 @@ class BBCCoUkIE(InfoExtractor):
                     })
             # Skip DASH until supported
             elif transfer_format == 'dash':
-                pass
+                formats.extend(self._extract_mpd_formats(
+                    href, programme_id, fatal=False))
             elif transfer_format == 'hls':
                 formats.extend(self._extract_m3u8_formats(
                     href, programme_id, ext='mp4', entry_protocol='m3u8_native',
```

**Comment by remitamine** [comment]
example pages:
kaltura i think they provide mpd formats for all videos.
for bbc i can get mpd formats for iplayer videos.
http://www.bbc.co.uk/iplayer/episode/b06zyl30
i will test more urls and post them here.

**Comment by remitamine** [comment]
> DASH formats should have lower priorities

i think we can explicitly lower the priority by changing the tbr into vbr for videos and abr for audios.
as the formats in mpd contain one of them an not both.

File: youtube_dl/extractor/common.py (hunk 2 above)
Comment (remitamine) [inline youtube_dl/extractor/common.py]: i changed it again to use `_add_ns`

File: youtube_dl/extractor/common.py (hunk 3 above)
Comment (yan12125) [inline youtube_dl/extractor/common.py]: ISO/IEC 23009-1:2014 declares the ```lang``` attribute uses rules defined in RFC5646, which then uses language codes from ISO639-2. ```mul```, ```und```, ```zxx``` and ```mis``` are special values. I guess they should be handled specially.

**Comment by yan12125** [comment]
> i think we can explicitly lower the priority by changing the tbr into vbr for videos and abr for audios

Not working for me. Maybe add an ```preference``` argument like ```_extract_m3u8_formats```?

**Comment by yan12125** [comment]
For ```test:kaltura```, ffmpeg fails in merging:
```
$ youtube-dl -v -f bestvideo+bestaudio test:kaltura --keep --prefer-ffmpeg
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-v', '-f', 'bestvideo+bestaudio', 'test:kaltura', '--keep', '--prefer-ffmpeg']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.02.01
[debug] Git HEAD: 1df4141
[debug] Python version 3.5.1 - Linux-4.4.0-4-ARCH-x86_64-with-arch-Arch-Linux
[debug] exe versions: avconv v12_dev0-2305-g77a44f5, avprobe v12_dev0-2305-g77a44f5, ffmpeg 2.8.6, ffprobe 2.8.6, rtmpdump 2.4
[debug] Proxy map: {}
[TestURL] Test URL: kaltura:269692:1_1jc2y3e4
[Kaltura] 1_1jc2y3e4: Downloading Kaltura signature
[Kaltura] 1_1jc2y3e4: Downloading video info JSON
[Kaltura] 1_1jc2y3e4: Downloading m3u8 information
[Kaltura] 1_1jc2y3e4: Downloading MPD manifest
WARNING: Requested formats are incompatible for merge and will be merged into mkv.
[debug] Invoking downloader on 'http://kalhlspd-a.akamaihd.net/dasha/p/269692/sp/26969200/serveFlavor/entryId/1_1jc2y3e4/v/12/pv/1/flavorId/0_,w4xtqo23,7eyew1f7,odf8oerb,np6oner9,/forceproxy/true/name/a.mp4.urlset/'
[download] Straight from the Heart-1_1jc2y3e4.ff4-v1.unknown_video has already been downloaded
[download] 100% of 43.74MiB
[debug] Invoking downloader on 'http://kalhlspd-a.akamaihd.net/dasha/p/269692/sp/26969200/serveFlavor/entryId/1_1jc2y3e4/v/12/pv/1/flavorId/0_,w4xtqo23,7eyew1f7,odf8oerb,np6oner9,/forceproxy/true/name/a.mp4.urlset/'
[download] Straight from the Heart-1_1jc2y3e4.ff4-a1.unknown_video has already been downloaded
[download] 100% of 3.82MiB
[ffmpeg] Merging formats into "Straight from the Heart-1_1jc2y3e4.mkv"
[debug] ffmpeg command line: ffmpeg -y -i 'file:Straight from the Heart-1_1jc2y3e4.ff4-v1.unknown_video' -i 'file:Straight from the Heart-1_1jc2y3e4.ff4-a1.unknown_video' -c copy -map 0:v:0 -map 1:a:0 'file:Straight from the Heart-1_1jc2y3e4.temp.mkv'
ERROR: file:Straight from the Heart-1_1jc2y3e4.ff4-v1.unknown_video: Invalid data found when processing input
Traceback (most recent call last):
  File "/home/yen/Executables/Multimedia/youtube-dl/youtube_dl/YoutubeDL.py", line 1736, in post_process
    files_to_delete, info = pp.run(info)
  File "/home/yen/Executables/Multimedia/youtube-dl/youtube_dl/postprocessor/ffmpeg.py", line 411, in run
    self.run_ffmpeg_multiple_files(info['__files_to_merge'], temp_filename, args)
  File "/home/yen/Executables/Multimedia/youtube-dl/youtube_dl/postprocessor/ffmpeg.py", line 155, in run_ffmpeg_multiple_files
    raise FFmpegPostProcessorError(msg)
youtube_dl.postprocessor.ffmpeg.FFmpegPostProcessorError: file:Straight from the Heart-1_1jc2y3e4.ff4-v1.unknown_video: Invalid data found when processing input
```

While avconv works fine:
```
$ youtube-dl -v -f bestvideo+bestaudio test:kaltura --keep
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-v', '-f', 'bestvideo+bestaudio', 'test:kaltura', '--keep']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.02.01
[debug] Git HEAD: 1df4141
[debug] Python version 3.5.1 - Linux-4.4.0-4-ARCH-x86_64-with-arch-Arch-Linux
[debug] exe versions: avconv v12_dev0-2305-g77a44f5, avprobe v12_dev0-2305-g77a44f5, ffmpeg 2.8.6, ffprobe 2.8.6, rtmpdump 2.4
[debug] Proxy map: {}
[TestURL] Test URL: kaltura:269692:1_1jc2y3e4
[Kaltura] 1_1jc2y3e4: Downloading Kaltura signature
[Kaltura] 1_1jc2y3e4: Downloading video info JSON
[Kaltura] 1_1jc2y3e4: Downloading m3u8 information
[Kaltura] 1_1jc2y3e4: Downloading MPD manifest
WARNING: Requested formats are incompatible for merge and will be merged into mkv.
[debug] Invoking downloader on 'http://kalhlspd-a.akamaihd.net/dasha/p/269692/sp/26969200/serveFlavor/entryId/1_1jc2y3e4/v/12/pv/1/flavorId/0_,w4xtqo23,7eyew1f7,odf8oerb,np6oner9,/forceproxy/true/name/a.mp4.urlset/'
[download] Destination: Straight from the Heart-1_1jc2y3e4.ff4-v1.unknown_video
[DashSegments] 1_1jc2y3e4: Downloading initialization segment
[DashSegments] 1_1jc2y3e4: Downloading segment 1 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 2 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 3 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 4 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 5 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 6 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 7 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 8 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 9 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 10 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 11 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 12 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 13 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 14 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 15 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 16 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 17 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 18 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 19 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 20 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 21 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 22 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 23 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 24 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 25 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 26 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 27 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 28 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 29 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 30 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 31 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 32 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 33 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 34 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 35 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 36 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 37 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 38 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 39 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 40 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 41 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 42 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 43 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 44 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 45 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 46 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 47 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 48 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 49 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 50 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 51 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 52 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 53 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 54 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 55 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 56 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 57 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 58 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 59 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 60 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 61 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 62 / 62
[download] 100% of 43.74MiB
[debug] Invoking downloader on 'http://kalhlspd-a.akamaihd.net/dasha/p/269692/sp/26969200/serveFlavor/entryId/1_1jc2y3e4/v/12/pv/1/flavorId/0_,w4xtqo23,7eyew1f7,odf8oerb,np6oner9,/forceproxy/true/name/a.mp4.urlset/'
[download] Destination: Straight from the Heart-1_1jc2y3e4.ff4-a1.unknown_video
[DashSegments] 1_1jc2y3e4: Downloading initialization segment
[DashSegments] 1_1jc2y3e4: Downloading segment 1 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 2 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 3 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 4 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 5 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 6 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 7 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 8 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 9 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 10 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 11 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 12 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 13 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 14 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 15 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 16 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 17 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 18 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 19 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 20 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 21 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 22 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 23 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 24 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 25 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 26 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 27 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 28 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 29 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 30 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 31 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 32 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 33 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 34 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 35 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 36 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 37 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 38 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 39 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 40 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 41 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 42 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 43 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 44 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 45 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 46 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 47 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 48 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 49 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 50 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 51 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 52 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 53 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 54 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 55 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 56 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 57 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 58 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 59 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 60 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 61 / 62
[DashSegments] 1_1jc2y3e4: Downloading segment 62 / 62
[download] 100% of 3.82MiB
[ffmpeg] Merging formats into "Straight from the Heart-1_1jc2y3e4.mkv"
[debug] ffmpeg command line: avconv -y -i 'file:Straight from the Heart-1_1jc2y3e4.ff4-v1.unknown_video' -i 'file:Straight from the Heart-1_1jc2y3e4.ff4-a1.unknown_video' -c copy -map 0:v:0 -map 1:a:0 'file:Straight from the Heart-1_1jc2y3e4.temp.mkv'
```
This may be a bug/missing feature in ffmpeg. Maybe we should report it to FFMpeg.

File: youtube_dl/extractor/common.py (hunk 3 above)
Comment (remitamine) [inline youtube_dl/extractor/common.py]: so we can just ignore this values?

Comment (yan12125) [inline youtube_dl/extractor/common.py]: Sounds OK.

**Comment by remitamine** [comment]
> Maybe add an preference argument like _extract_m3u8_formats?

yes, i will add it.

**Comment by remitamine** [comment]
i will compile libav and test it also with bbc video dash formats may be it can handle them.

**Comment by remitamine** [comment]
test:kaltura_3 give me a simular message to what i get in bbc:
```
ffprobe Kaltura\ Player\ ToolKit-1_sf5ovm7u.mp4 
ffprobe version 2.8.6 Copyright (c) 2007-2016 the FFmpeg developers
  built with gcc 5.3.0 (GCC)
  configuration: --prefix=/usr --disable-debug --disable-static --disable-stripping --enable-avisynth --enable-avresample --enable-fontconfig --enable-gnutls --enable-gpl --enable-ladspa --enable-libass --enable-libbluray --enable-libdcadec --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libmodplug --enable-libmp3lame --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-libschroedinger --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libv4l2 --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-shared --enable-version3 --enable-x11grab
  libavutil      54. 31.100 / 54. 31.100
  libavcodec     56. 60.100 / 56. 60.100
  libavformat    56. 40.101 / 56. 40.101
  libavdevice    56.  4.100 / 56.  4.100
  libavfilter     5. 40.101 /  5. 40.101
  libavresample   2.  1.  0 /  2.  1.  0
  libswscale      3.  1.101 /  3.  1.101
  libswresample   1.  2.101 /  1.  2.101
  libpostproc    53.  3.100 / 53.  3.100
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x5558e0750700] multiple fourcc not supported
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x5558e0750700] Could not find codec parameters for stream 0 (Video: none (encv / 0x76636E65), none, 1280x720, 1447 kb/s): unknown codec
Consider increasing the value for the 'analyzeduration' and 'probesize' options
Kaltura Player ToolKit-1_sf5ovm7u.mp4: Operation not permitted
```
```
avprobe Kaltura\ Player\ ToolKit-1_sf5ovm7u.mp4 
avprobe version v12_dev0-2307-g1ba1fed, Copyright (c) 2007-2015 the Libav developers
  built on Feb  3 2016 15:06:55 with gcc 5.3.0 (GCC)
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x22eb8c0] multiple fourcc not supported
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x22eb8c0] max_analyze_duration 5000000 reached
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'Kaltura Player ToolKit-1_sf5ovm7u.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 1
    compatible_brands: isomavc1
  Duration: 00:01:54.28, start: 5.360000, bitrate: 1526 kb/s
    Stream #0:0(und): Video: encv / 0x76636E65 [encv / 0x76636E65]
      none, 1280x720, 1448 kb/s
      24 fps, 12800 tbn (default)
    Metadata:
      handler_name    : VideoHandler
Unsupported codec with id 0 for input stream 0
# avprobe output
```

**Comment by remitamine** [comment]
there is a problem with parse_duration sometimes it can convert some date strings like this `PT1H0.040S`

**Comment by remitamine** [comment]
bbc dash formats can be played with gpac MP4Client i will open an issue in both ffmpeg and libav trac.

**Comment by yan12125** [comment]
```parse_duration``` should be fixed/extended to handle it.

And as a record, I've opened a ticket for Kaltura DASH video: https://trac.ffmpeg.org/ticket/5216.

**Comment by remitamine** [comment]
about parse_duration, it will necessary to fix it when we add support for bbc and kaltura because the duration is needed only for SegmentTemplate without a SegmentTimeline.
i also tested the code with mpds from http://rdmedia.bbc.co.uk/.
it can handle most of them(with the latest changes).

**Comment by remitamine** [comment]
i received about the problem for BBC videos with ffmpeg.
it should be fixed, but it will take time to be released in different platforms.
i think it's better to keep adding support for kaltura and bbc untill the problems with ffmpeg and libav fixed.

is there any thing else before merging this PR.

**Comment by yan12125** [comment]
FFMpeg 3.0 is released [1]. I think the BBC part can be pushed to master.

[1] http://phoronix.com/scan.php?page=news_item&px=FFmpeg-3.0-Released

**Comment by yan12125** [comment]
Kaltura errors fixed in ffmpeg git: https://trac.ffmpeg.org/ticket/5216#comment:6

**Comment by remitamine** [comment]
sorry, i foreget to replay about this comment
> FFMpeg 3.0 is released [1]. I think the BBC part can be pushed to master.

i didn't push it to master because many linux distrubution doesn't ship the version 3: http://pkgs.org/download/ffmpeg.

> Kaltura errors fixed in ffmpeg git: https://trac.ffmpeg.org/ticket/5216#comment:6

this is good, when it released and shipped, we will push the patch to enable DASH in Kaltura.

**Comment by yan12125** [comment]
Thanks for the information. http://pkgs.org is quite useful.

My comments are just remainders. BBC and Kaltura seem not providing better quality videos in DASH manifests like Facebook and YouTube, so these changes are not critical.

**Pull request merged**